drivers: sensor: Update drivers to use devicetree Kconfig symbol

galak opened the pull request.

galak committed:
drivers: sensor: Update drivers to use devicetree Kconfig symbol

Update sensor drivers to use DT_HAS_<compat>_ENABLED Kconfig symbol
to expose the driver and enable it by default based on devicetree.

Signed-off-by: Kumar Gala <galak@kernel.org>

galak committed:
tests: build_all: sensor: remove unnecessary Kconfig symbols

Kconfig will default to enable the drivers based on devicetree so
we don't need to explicitly set the Kconfig symbols.

Signed-off-by: Kumar Gala <galak@kernel.org>

bbilas commented:
@galak Are you gonna also remove the unnecessary Kconfig symbols from samples? For example https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/sensor/ina219/prj.conf#L5=

galak commented:
> @galak Are you gonna also remove the unnecessary Kconfig symbols from samples? For example https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/sensor/ina219/prj.conf#L5=

Not in this PR.  Happy to have help from others on that.

MaureenHelm commented on drivers/sensor/adxl345/Kconfig:
```diff
@@ -7,6 +7,7 @@ config ADXL345
 	default y
 	depends on DT_HAS_ADI_ADXL345_ENABLED
-	depends on I2C || SPI
+	depends on (I2C && $(dt_compat_on_bus,$(DT_COMPAT_ADI_ADXL345),i2c)) || \
+		   (SPI && $(dt_compat_on_bus,$(DT_COMPAT_ADI_ADXL345),spi))
```
I don't quite understand this logic. The sensor has to be on one type of bus or the other, so isn't it redundant to qualify with `dt_compat_on_bus` now that it depends on `DT_HAS_ADI_ADXL345_ENABLED`?

galak replied:
No, there was a case that the sensor was on I2C and enabled, but CONFIG_I2C=n and CONFIG_SPI=y.  So `DT_HAS_ADI_ADXL345_ENABLED` will be 'y', so we need to qualify such a case (which is why that mess of logic).

galak committed:
boards: arm: sensortile_box: Configure I2C if sensors

If a test or sample enables CONFIG_SENSOR=y than any sensor on the
I2C or SPI bus should be available.  We already handle this for
SPI on the board, add I2C as well.

Signed-off-by: Kumar Gala <galak@kernel.org>
